app_dart/test/service/github_checks_service_test.dart

```diff
@@ -0,0 +1,203 @@
+              as Map<String, dynamic>);
+      await githubChecksService.handleCheckSuite(
+        checkSuiteEvent,
+        mockLuciBuildService,
+      );
+      expect(
+          verify(mockLuciBuildService.rescheduleUsingCheckSuiteEvent(
+                  captureAny, captureAny))
+              .captured,
+          <dynamic>[
+            checkSuiteEvent,
+            checkRun,
+          ]);
+    });
+  });
+  group('handleCheckRunEvent', () {
+    test('rerequested triggers triggers a luci build', () async {
+      final CheckRunEvent checkRunEvent = CheckRunEvent.fromJson(
+        jsonDecode(checkRunString) as Map<String, dynamic>,
+      );
+      await githubChecksService.handleCheckRun(
+        checkRunEvent,
+        mockLuciBuildService,
+      );
+      expect(
+          verify(mockLuciBuildService.rescheduleUsingCheckRunEvent(
+            captureAny,
+          )).captured,
+          <dynamic>[checkRunEvent]);
+    });
+  });
+  group('updateCheckStatus', () {
+    test('Userdata is empty', () async {
+      final push_message.BuildPushMessage buildMessage =
+          push_message.BuildPushMessage.fromJson(
+              jsonDecode(buildPushMessageJsonTemplate(''))
+                  as Map<String, dynamic>);
+      final bool success = await githubChecksService.updateCheckStatus(
+          buildMessage, mockLuciBuildService, slug);
+      expect(success, isFalse);
+    });
+    test('Userdata does not contain check_run_id', () async {
+      final push_message.BuildPushMessage buildMessage =
+          push_message.BuildPushMessage.fromJson(
+              jsonDecode(buildPushMessageJsonTemplate('{\\"retries\\": 1}'))
+                  as Map<String, dynamic>);
+      final bool success = await githubChecksService.updateCheckStatus(
+          buildMessage, mockLuciBuildService, slug);
+      expect(success, isFalse);
+    });
+    test('Userdata contain check_run_id', () async {
+      when(mockGithubChecksUtil.getCheckRun(any, any, any))
+          .thenAnswer((_) async => checkRun);
+      final push_message.BuildPushMessage buildPushMessage =
+          push_message.BuildPushMessage.fromJson(
+              jsonDecode(buildPushMessageJsonTemplate('{\\"check_run_id\\": 1,'
+                  '\\"repo_owner\\": \\"flutter\\",'
+                  '\\"repo_name\\": \\"cocoon\\"}')) as Map<String, dynamic>);
+      await githubChecksService.updateCheckStatus(
+          buildPushMessage, mockLuciBuildService, slug);
+      expect(
+          verify(mockGithubChecksUtil.updateCheckRun(
+            any,
+            any,
+            captureAny,
+            status: anyNamed('status'),
+            conclusion: anyNamed('conclusion'),
+            detailsUrl: anyNamed('detailsUrl'),
+            output: anyNamed('output'),
+          )).captured,
+          <github.CheckRun>[checkRun]);
+    });
+  });
+}
+
+String buildPushMessageJsonTemplate(String jsonUserData) => '''{
```

Comment:
This is hard to change, is it possible to construct push_message.BuildPushMessage directly?

Reply:
It may be possible but the end solution may be equally complex as this object requires several layers of objects all of them available at compilation time. I tried creating a build model using the constructor and soon I got an error telling me  that const instances could not be used to create the object after trying for several hours the only reliable way I found to create these objects was using the fromJson.

Reply:
Instead of compilation time, this could be a function returning an object at runtime.

Reply:
Sounds good, I'll give it a try and if I make it work I'll send a new PR to refactor the jsons in the different classes.